pkg/controller/apmserver/apmserver_controller.go

```diff
@@ -180,29 +178,34 @@ func (r *ReconcileApmServer) Reconcile(request reconcile.Request) (reconcile.Res
 
 	var as apmv1beta1.ApmServer
 	if ok, err := association.FetchWithAssociation(r.Client, request, &as); !ok {
```

Comment:
This does not come from your PR, but I just realized the bool + err returns seems a bit unusual, especially since we only check `ok` and not `err`.

I guess a more common pattern would be:

```go

if err := association.FetchWithAssociation(r.client, request, &as); err != nil {
    if apierrors.IsNotFound(err) {
        // association was deleted
        r.OnDelete(types.NamespacedName{ns, n})
        return reconcile.Result{}, nil
    }
    return err
}
```

The bool feels a bit useless.

Reply:
👍  Yeah, you're right. Since it has an impact on some unit tests I would like to address it in a different PR, this one is already pretty large.

Reply:
Sounds good!